src/main/scala/bench/Benchmark.scala

```diff
@@ -0,0 +1,239 @@
+package bench
+
+import strawman.collection.{immutable, mutable}
+import strawman.collection.immutable.{LazyList, List}
+import java.lang.{Object, String}
+
+import org.scalameter.picklers.Implicits.stringPickler
+import org.scalameter._
+import strawman.collection.mutable.{ArrayBuffer, ListBuffer}
+
+import scala.{Any, Int, Unit}
+import scala.Predef.identity
+
+// You can run the benchmarks with `sbt run`
+// Comment the following `include(…)` lines to exclude a benchmark in particular
+// These benchmarks have been heavily inspired by those: https://github.com/lihaoyi/scala-bench
+object Benchmark extends Bench.Group {
+
+  include(new Cons {})
+  include(new Uncons {})
+  include(new Concat {})
+  include(new Foreach {})
+  include(new Lookup {})
+  include(new MemoryFootprint {})
+
+}
+
+trait MemoryFootprint extends Bench.ForkedTime with Generators {
+
+  override def measurer = new Executor.Measurer.MemoryFootprint
+
+  def benchmark[A](name: String, generator: Gen[A]): Unit =
+    performance of name in {
+      measure method "memory-footprint" in {
+        using(generator).curve("memory-footprint") in identity
+      }
+    }
+
+  benchmark("scala.List", scalaLists)
+  benchmark("List", lists)
+  benchmark("LazyList", lazyLists)
+  benchmark("ArrayBuffer", arrayBuffers)
+  benchmark("ListBuffer", listBuffers)
+
+}
+
+trait Cons extends Bench.ForkedTime with Generators {
```

Comment:
I would say that, for micro-benchmarks like those I think it is better to ignore GCs (though for a more complex benchmark like a complete sorting algorithm, maybe it would be better to not ignore GCs). What do you think?